storybook/sidebar.js

```diff
@@ -41,7 +43,7 @@ function useIcons( item ) {
 	return useMemo( () => {
 		let data = {};
 
-		if ( item.isComponent && item.children?.length ) {
+		if ( item.type === 'component' && item.children?.length ) {
```

Comment:
I didn't see this change documented in the changelog, but without it, the badges won't appear - `isComponent` doesn't exist, but I saw `type` is `component`, so we're using that.